### PR TITLE
Bluetooth: Mesh: no more tinycrypt in ble mesh tfm images

### PR DESCRIPTION
**Pull request by alxelax**
PR allows to get rid of tinycrypt objects from
the final binary of the ble mesh apps based on
PSA TFM crypto.

Final size of mesh app for nrf5340dk_nrf5340_cpuapp_ns (that is build with TFM PSA crypto)
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      120936 B       192 KB     61.51%
             RAM:       35350 B       192 KB     17.98%
        IDT_LIST:          0 GB         2 KB      0.00%
```

without these changes tinycrypt sneaks into it
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      126120 B       192 KB     64.15%
             RAM:       37294 B       192 KB     18.97%
        IDT_LIST:          0 GB         2 KB      0.00%
```
+
some bsim tests fixes due to collisions after timing changes.

### CODE REVIEW

File: subsys/bluetooth/host/Kconfig [hunk 1]
```diff
@@ -163,12 +163,15 @@ rsource "../mesh/Kconfig"
...
-	# library if this is not provided by the controller implementation.
-	bool
+	bool "Use crypto functionality implemented in the Bluetooth host"
 	default y if !BT_CTLR_CRYPTO
 	select TINYCRYPT
 	select TINYCRYPT_AES
+	help
+	  The option adds the AES encryption support using TinyCrypt
+	  library if this is not provided by the controller implementation.
+	  Either the option is not required if BLE Mesh is enabled and
```
Comment (akredalen) [inline subsys/bluetooth/host/Kconfig:173]:
```suggestion
	  The option is not required if BLE Mesh is enabled and
```

File: samples/bluetooth/mesh/boards/nrf5340dk_nrf5340_cpuapp_ns.conf [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# The option adds tinycrypt based bt_rand.
```
Comment (akredalen) [inline samples/bluetooth/mesh/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:1]:
```suggestion
# The option adds TinyCrypt based bt_rand.
```
Same for three other .conf-files.

File: samples/bluetooth/mesh/boards/nrf5340dk_nrf5340_cpuapp_ns.conf [hunk 3]
```diff
@@ -1,3 +1,8 @@
...
+CONFIG_BT_HOST_CRYPTO=n
+# The option adds gatt caching feature that is based on tinycrypt.
```
Comment (akredalen) [inline samples/bluetooth/mesh/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:3]:
```suggestion
# The option adds GATT caching feature that is based on TinyCrypt.
```
Same for three other .conf-files.

**Review by akredalen — CHANGES_REQUESTED** [review]

**Review by akredalen — DISMISSED** [review]

File: subsys/bluetooth/host/Kconfig [hunk 4]
```diff
@@ -163,12 +163,15 @@ rsource "../mesh/Kconfig"
...
-	# library if this is not provided by the controller implementation.
-	bool
+	bool "Use crypto functionality implemented in the Bluetooth host"
 	default y if !BT_CTLR_CRYPTO
 	select TINYCRYPT
 	select TINYCRYPT_AES
+	help
+	  The option adds the AES encryption support using TinyCrypt
+	  library if this is not provided by the controller implementation.
+	  The option is not required if BLE Mesh is enabled and
```
Comment (PavelVPV) [inline subsys/bluetooth/host/Kconfig:173]: Not sure if mesh should be mentioned here. What about `bt_encrypt_le` and `bt_encrypt_be` functions? They aren't replaced when mesh is enabled with PSA support. This options still needs to be enabled if a customer wants to enable SMP (Security Manager Protocol, not SMP from MCUmgr).

File: samples/bluetooth/mesh_provisioner/boards/nrf5340dk_nrf5340_cpuapp_ns.conf [hunk 5]
```diff
@@ -1,3 +1,8 @@
+# The option adds TinyCrypt based bt_rand.
+CONFIG_BT_HOST_CRYPTO=n
+# The option adds GATT caching feature that is based on TinyCrypt.
+CONFIG_BT_GATT_CACHING=n
```
Comment (PavelVPV) [inline samples/bluetooth/mesh_provisioner/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:4]: Is `CONFIG_BT_GATT_CACHING=n` needed here? mesh_provisioner doesn't enable neither peripheral nor central role support.

File: samples/bluetooth/mesh/boards/nrf5340dk_nrf5340_cpuapp_ns.conf [hunk 6]
```diff
@@ -1,3 +1,8 @@
+# The option adds TinyCrypt based bt_rand.
+CONFIG_BT_HOST_CRYPTO=n
+# The option adds GATT caching feature that is based on TinyCrypt.
+CONFIG_BT_GATT_CACHING=n
```
[the remark this answers is not in the document]
Comment (alxelax) [reply samples/bluetooth/mesh/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:4]: I guess this is not sufficient. Host also supports ecc, aes ccm and randomization with tinycrypt. Host requires significant refactoring to get rid of tinycrypt. This is quite huge work that we've already done for mesh.

File: subsys/bluetooth/host/Kconfig (hunk 4 above)
Comment (alxelax) [reply subsys/bluetooth/host/Kconfig:173]: Mesh doesn't use them. Yes, if customer adds anything that uses them, tinycrypt will appear again. I guess there is no common strategy to get rid tinycrypt in Zephyr in general.

**Review by PavelVPV — APPROVED** [review]

**Review by akredalen — APPROVED** [review]

**Pull request merged**